src/Microsoft.ServiceFabric.Services.Remoting/Client/ServiceProxyFactory.cs

```diff
@@ -91,65 +91,45 @@ public ServiceProxyFactory(OperationRetrySettings retrySettings = null)
         {
             var serviceInterfaceType = typeof(TServiceInterface);
 
-#if !DotNetCoreClr
-
             //Use provider to find the stack
-            if (this.proxyFactoryV1 == null && this.proxyFactoryV2 == null)
```

Comment:
Why this check? This way we break interop for wherer you need orchestration between V1 and V2 services. Either V1 factory is created, or V2 -- never both.

Reply:
Its just checking if none of the V1 and V2 factory is provided,c reate from the attribute.

Reply:
Its already possible to talk to a V1 or V2 listener https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-services-communication-remoting#how-to-upgrade-from-remoting-v1-to-remoting-v2

Reply:
I think there's some confusion here, due to me not explaining why I've made these changes. `ServiceProxy` static methods use static instance of `ServiceProxyFactory`, if we use `ServiceProxy` to orchestrate between v1 and v2 services, you can not, as the internal DefaultProxyFactory will be initialised to either a v1 factory or a v2 factory. See below comment also!

Reply:
Thanks for explaining, yes, i see your point when you want to use the same factory to talk to one service with V1 and then talk to other service with V2. When this was designed, it assumed one factory for V1 and other for V2.

Reply:
That would make sense! You can close this PR if you like, unless you want me to do anything with it?

If V1 stack is being deprecated in the near future, I assume you have some sort of plan for the ServiceProxy class, respective factories (and similarly for Actors). It would seem at this point, any change would be a breaking change.

I'd perhaps deprecate `Microsoft.ServiceFabric.Services.Remoting.Client.ServiceProxyFactory`. Move the `IService` provider (stack selection) logic to `ServiceProxy.Create<>()` and have this use the appropriate default static instance of `Remoting.V2.Client.ServiceProxyFactory` (or v1 if not using netcore). I have a very naive view of SF inner workings though, so this may be a bad idea!

Thanks again for all your responses!

Reply:
Sure @Mardoxx . We would try to deprecate V1 in a non breaking way after the customers have moved the services and client to v2 stack. Thanks!.